internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -223,11 +224,22 @@ func (md *machineDeployment) deployCreateMachinesForGroups(ctx context.Context,
 		}
 
 		services := groupConfig.AllServices()
+		autostopSetting := fly.MachineAutostopSuspend
```

Comment:
why isn't `fly.MachineAutostopOff` the default? below logic doesn't play well when there are no services; it will default to include the group in groupsWithAutosuspendEnabled.

I don't know follow why you insist in using a out-of-loop variable (the `autostopSetting`) instead of something like:

```go
for _, s := range services {
  switch v := s.AutostopMachines; {
  case v == nil || *v == fly.MachineAutostopOff:
    // skip
  case *v == fly.MachineAutostopStop:
    groupsWithAutostopEnabled[name] = true
  case *v == fly.MachineAutostopSuspend:
    groupsWithAutosuspendEnabled[name] = true
  }
}

Reply:
Agh, thanks, I goofed this again!

I'm trying to take the minimum of the autostop settings across all of the process group's services, because that is what the proxy will use to decide how to downscale the group. E.g. if the process group has three services, two with `auto_stop_machines = true` and one with `auto_stop_machines = false`, then the proxy will take no action because of the third, not-autostoppable service. But (if I understand correctly) the existing logic would print that autostop is enabled anyway.